outrank/algorithms/neural/mlp_nn.py

```diff
@@ -0,0 +1,136 @@
+from __future__ import annotations
+
+import logging
+
+import jax
+import optax
+from flax import linen as nn
+from jax import numpy as jnp
+from jax import random
+
+logger = logging.getLogger('syn-logger')
+logger.setLevel(logging.DEBUG)
+
+key = random.PRNGKey(7235123)
+
+
+class GenericJaxNN(nn.Module):
+    num_features: int
+    architecture: jnp.array
+
+    @nn.compact
+    def __call__(self, x):
+        for num_units in self.architecture:
+            x = nn.Dense(features=num_units)(x)
+            x = nn.relu(x)
+            x = nn.Dense(features=2)(x)
+        return x
+
+
+class NNClassifier:
+
+    def __init__(self, learning_rate=0.001, architecture=[48, 48], epochs=100):
+        self.learning_rate = learning_rate
+        self.architecture = architecture
+        self.num_epochs = epochs
+        self.ncl = None
+
+        batch_size = 10
+        features = 5  # Number of input features to the model
+
+        # Pass the features and architecture parameters to the model constructor
+        self.mlp_model = GenericJaxNN(
+            num_features=features,
+            architecture=jnp.array(architecture),
+        )
+
+    def fit(self, X, Y, print_loss=True):
+
+        loss_grad_fn = jax.value_and_grad(self.forward_loss)
+        self.ncl = len(jnp.unique(X))
+        X = jax.nn.one_hot(X, num_classes=self.ncl).reshape(X.shape[0], -1)
+        sample_batch = jnp.ones((1, X.shape[1]))
```

Comment:
Did you test perf with other initializers? Not sure how well that behaves with a hardcoded PRNGKey.

Reply:
It shouldn't and it doesn't matter for a somewhat ok space of categorical problems we're considering here.